tests/src/python/test_qgsgeometryvalidator.py

```diff
@@ -74,6 +74,50 @@ def test_linestring_duplicate_nodes(self):
         self.assertEqual(spy[2][0].where(), QgsPointXY(1, 1))
         self.assertEqual(spy[2][0].what(), 'line 1 contains 3 duplicate node(s) starting at vertex 1')
 
+    def test_linestring_intersections(self):
+        # no intersections
+        g = QgsGeometry.fromWkt("LineString (0 0, 10 0, 10 10, 0 10)")
+        validator = QgsGeometryValidator(g)
+        spy = QSignalSpy(validator.errorFound)
+        validator.run()
+
+        self.assertEqual(len(spy), 0)
+
+        # accept closed linestring
+        g = QgsGeometry.fromWkt("LineString (0 0, 10 0, 10 10, 0 10, 10 10)")
```

Comment:
```suggestion
        g = QgsGeometry.fromWkt("LineString (0 0, 10 0, 10 10, 0 10, 0 0)")
```
The current geom should be failing?

Reply:
Aaah, I fell asleep on the d key of my vim and deleted part of the test... 😓 

The proposed geometry is valid, the tool doesn't detect overlapping lines (it's neither a duplicate point, nor an "intersection"). And indeed, your geometry is the one that matches the description. I've force-pushed the test.

Reply:
Does the existing code also allow overlapping lines? (Seems like a seperate bug we should address, overlapping lines should make a geometry invalid 🤪)

Reply:
Yes, the actual code also allows overlapping lines.
As mentioned in my test comment, this case is "ambiguous" ; at the moment, idk if it's a bug or not. I'm reading norm and seeing what is done in other software.
Furthermore, I always think that this kind of "validity" should be more nuanced because what may be valid for one case (here, an example of an electrical cable that goes back and forth: so overlap is okay here), is not valid for another. Similarly, one could very well consider it's invalid to have duplicates, but self-intersection is acceptable. But that's another PR :)